cmd/juju/status.go

```diff
@@ -203,8 +203,11 @@ func (s serviceStatus) GetYAML() (tag string, value interface{}) {
 }
 
 type unitStatus struct {
-	Err            error                 `json:"-" yaml:",omitempty"`
-	Charm          string                `json:"upgrading-from,omitempty" yaml:"upgrading-from,omitempty"`
+	Err                error              `json:"-" yaml:",omitempty"`
+	Charm              string             `json:"upgrading-from,omitempty" yaml:"upgrading-from,omitempty"`
+	WorkloadStatusInfo statusInfoContents `json:"workload-status,omitempty" yaml:"workload-status,omitempty"`
```

Comment:
To make this easy for people to understand, please add comments to indicate what AgentStatusInfo vs AgentState vs AgentStateInfo etc are. There's so many different ones now, all very similar and not obvious without comments what else one is. Even something like statusInfoContents below could maybe be commented.